src/views/Swap.vue

```diff
@@ -675,33 +675,39 @@ export default {
       }
 
       if (this.availableFees.has(this.assetChain)) {
-        const feePrice = this.getAssetFees(this.assetChain)[
+        const fromTxTypes = this.getFeeTxTypes(this.assetChain)
+        const fromAssetFee = this.getAssetFees(this.assetChain)[
           this.selectedFee[this.assetChain]
         ].fee
-        const initiationFee = getTxFee(
-          this.asset,
-          TX_TYPES.SWAP_INITIATION,
-          feePrice
-        )
-        fees[this.assetChain] = initiationFee
+
+        const fromFee = fromTxTypes.reduce((accum, tx) => {
+          return accum.plus(getTxFee(this.asset, tx, fromAssetFee))
+        }, BN(0))
+
+        fees[this.assetChain] = fromFee
       }
 
       if (this.availableFees.has(this.toAssetChain)) {
-        const feePrice = this.getAssetFees(this.toAssetChain)[
+        const toTxTypes = this.getFeeTxTypes(this.toAssetChain)
+        const toAssetFee = this.getAssetFees(this.toAssetChain)[
           this.selectedFee[this.toAssetChain]
         ].fee
-        const claimFee = getTxFee(this.toAsset, TX_TYPES.SWAP_CLAIM, feePrice)
-        fees[this.toAssetChain] = fees[this.toAssetChain]
-          ? fees[this.toAssetChain].plus(claimFee)
-          : claimFee
+        console.log('toTxTypes', toTxTypes)
+        const toFee = toTxTypes.reduce((accum, tx) => {
+          return accum.plus(getTxFee(this.toAsset, tx, toAssetFee))
+        }, BN(0))
+
+        fees[this.toAssetChain] = toFee
 
         if (this.sendTo) {
-          const sendFee = getTxFee(this.toAsset, TX_TYPES.SEND, feePrice)
+          const sendFee = getTxFee(this.toAsset, TX_TYPES.SEND, toAssetFee)
```

Comment:
Is sendto fee being accounted for twice?
It looks like this would already be included because on `toFee` is also accumulates toTxTypes = `TX_TYPES.SEND`

Reply:
got it, let me fix it